### ISSUE TITLE
iconv does not work properly on fpm-alpine

### ISSUE THREAD

**Issue by TheTimeWalker** [nextcloud/docker#1299]
There currently seems to be an issue with iconv on alpine linux which means that specific plugins and other NC functionalities break if iconv is needed. This happens for example with the mail app and is noted in this issue: https://github.com/nextcloud/mail/issues/3999

There seems to be apossible workaround as linked to the corresponding StackOverflow link: https://stackoverflow.com/questions/63746822/why-does-iconv-returns-empty-string-in-php7-4-fpm-alpine-docker/63747282#63747282
However I'm not sure if this is desirable to be implemented in this Dockerfile

**Comment by sfornengo** [comment]
This issue has been killing the mail application for weeks...
How to switch back from nextcloud:fpm-alpine to nexcloud:fpm ?
Only replacing nextcloud:fpm-alpine by nextcloud:fpm in docker-compose.yml doesn't work for me...
Could someone writes the steps to achieve this switch ?

**Comment by snevas** [comment]
@sfornengo For me it was as simple as changing the image in the docker-compose. But please keep the sequence in mind:
- docker-compose down
- remove remnants you find with `docker ps -a`
- change docker-compose.yml image line
- docker-compose up

**Comment by sfornengo** [comment]
@snevas, here is the result:
![image](https://user-images.githubusercontent.com/1094525/103011744-b7dbff00-453a-11eb-84bc-eceb0aa45d53.png)
Any idea ?

**Comment by sfornengo** [comment]
Ok, switching from fpm-alpine to fpm need also to chown /var/www to remove the error below:
docker exec nextcloud-docker_app_1 chown -R www-data:root /var/www

But even under fpm instead of fpm-alpine the mail app stucks on "loading messages"...
I thought the problem was iconv only under fpm-alpine...
Could someone help ?

**Comment by almereyda** [comment]
An exemplary fix for the linuxservers image is displayed in https://github.com/linuxserver/docker-nextcloud/pull/166

This is much needed for https://github.com/nextcloud/mail/issues/4351

For those already building their Nextcloud image from the examples in this repository themselves, adding this to the `app` `Dockerfile` is what helped me to work around the issue:

```
RUN set -ex; \
  \
  apk add --no-cache \
    gnu-libiconv \
    php7-iconv \
  ;
ENV LD_PRELOAD=/usr/lib/preloadable_libiconv.so
```

Indeed if it's mandatory for the Alpine platform, we might want to consider to include these packages in the Dockerfile by default.

**Comment by J0WI** [comment]
We use the PHP Docker image. `php7-iconv` pulls the PHP packages from the Alpine repo.
See https://github.com/docker-library/php/issues/1121

**Comment by AlterDepp** [comment]
Fixed for me with nextcloud 20.0.8-fpm-alpine and mail app 1.9.1
Thanks
